modules/core/src/main/kotlin/datamaintain/core/step/check/rules/implementations/ExecutedScriptsNotRemovedCheck.kt

```diff
@@ -0,0 +1,36 @@
+package datamaintain.core.step.check.rules.implementations
+
+import datamaintain.core.script.ExecutedScript
+import datamaintain.core.script.ScriptWithContent
+import datamaintain.core.step.check.rules.ScriptType
+import datamaintain.core.step.check.rules.contracts.FullContextCheckRule
+import java.lang.IllegalStateException
+
+class ExecutedScriptsNotRemovedCheck(
+        executedScripts: Sequence<ExecutedScript>
+) : FullContextCheckRule(executedScripts) {
+    override fun check(scripts: Sequence<ScriptWithContent>) {
+        val executedScriptChecksumsNotFoundInSortedScripts = executedScripts
+                .map { it.checksum }
+                .minus(scripts.map { it.checksum })
+                .toList()
+
+        if (executedScriptChecksumsNotFoundInSortedScripts.isNotEmpty()) {
```

Comment:
Instead of casting the sequence to a list, you may use ```count()``` and check that the count equals 0

Reply:
No, I want to log the list of differences.